mteb/models/openai_models.py

```diff
@@ -15,27 +15,49 @@
 
 
 class OpenAIWrapper(Wrapper):
-    def __init__(self, model_name: str, embed_dim: int | None = None, **kwargs) -> None:
+    def __init__(
+        self,
+        model_name: str,
+        max_seq_length: int,
+        tokenizer_name: str = "cl100k_base",  # since all models use this tokenizer now
+        embed_dim: int | None = None,
+        **kwargs,
+    ) -> None:
         requires_package(self, "openai", "Openai text embedding")
         from openai import OpenAI
 
         self._client = OpenAI()
         self._model_name = model_name
         self._embed_dim = embed_dim
+        self._max_seq_length = max_seq_length
+        self._tokenizer_name = tokenizer_name
 
     def encode(self, sentences: list[str], **kwargs: Any) -> np.ndarray:
         requires_package(self, "openai", "Openai text embedding")
+        import tiktoken
```

Comment:
You need to check if `tiktoken` library is installed. You can use `requires_package` function`